zerver/lib/events.py

```diff
@@ -242,6 +243,12 @@ def fetch_initial_state_data(
             include_deactivated_groups=include_deactivated_groups,
             fetch_anonymous_group_membership=True,
         )
+        anonymous_group_membership_data_dict: dict[int, UserGroupMembersDataDict] = {}
+        for key, value in realm_groups_data.anonymous_group_membership.items():
+            anonymous_group_membership_data_dict[key] = UserGroupMembersDataDict(
+                direct_members=value["direct_members"],
+                direct_subgroups=value["direct_subgroups"],
+            )
```

Comment:
We turn this back into a data dict, which feels a bit weird to do. But I think it's better to do this, since if we switch to anonymous groups using typed dicts, a lot of the current data dicts will be replaced by typedDicts